client/components/domains/reskin-side-explainer/index.jsx

```diff
@@ -1,28 +1,45 @@
+import { isEnabled } from '@automattic/calypso-config';
 import { localize } from 'i18n-calypso';
 import { Component } from 'react';
+import { connect } from 'react-redux';
+import { isEligibleForProPlan } from 'calypso/my-sites/plans-comparison';
+import { getSelectedSiteId } from 'calypso/state/ui/selectors';
 
 import './style.scss';
 
 class ReskinSideExplainer extends Component {
 	getStrings() {
-		const { type, translate } = this.props;
+		const { type, eligibleForProPlan, selectedSiteId, translate } = this.props;
 
 		let title;
 		let subtitle;
 		let ctaText;
 
+		// The special case latter is for handling the case in the sign-up flow.
+		// By that time, a site is not available yet, so we can only check the feature flag for now
+		const shouldShowProPlanTitle =
+			eligibleForProPlan || ( selectedSiteId == null && isEnabled( 'plans/pro-plan' ) );
```

Comment:
Non blocking, but since the `isEligibleForProPlan` only/mostly [checks](https://github.com/Automattic/wp-calypso/blob/trunk/client/my-sites/plans-comparison/is-eligible-for-pro-plan.ts#L21) for `isEnabled( 'plans/pro-plan' )`, the second condition seems redundant.
On the other hand, this might be safer in case extra conditions will be added to `isEligibleForProPlan` in the future.